returnn/torch/engine.py

```diff
@@ -611,8 +611,12 @@ def _create_data_loader(self, dataset: Dataset) -> DataLoader:
                 wrapped_dataset, chunking, min_chunk_size=min_chunk_size
             )
 
-        assert self.config.typed_value("batch_size") is not None, "batch_size not defined in config"
+        assert self.config.typed_value("batch_size") is not None or (
+            self.config.typed_value("batch_size_train") is not None
+            and self.config.typed_value("batch_size_dev") is not None
+        ), "batch_size not defined in config"
```

Comment:
there could be cases e.g. forwarding only where it would be ok to only set `batch_size_dev` and not set (`batch_size` and `batch_size_train`). Is the assert like this ok or should we rather do

```
batch_size = self.config.typed_value("batch_size", -1)
batch_size = self.config.typed_value(f"batch_size_{'train' if train else 'dev'}", batch_size)
assert batch_size != -1, "batch_size not defined in config"

Reply:
Yeah, `assert batch_size != -1` is probably easier. It should also report the `train` flag then.